log4j-core/src/main/java/org/apache/logging/log4j/core/impl/Log4jLogEvent.java

```diff
@@ -91,12 +91,8 @@ public Builder() {
 
         public Builder(final LogEvent other) {
             Objects.requireNonNull(other);
-            if (other instanceof RingBufferLogEvent) {
-                ((RingBufferLogEvent) other).initializeBuilder(this);
-                return;
-            }
-            if (other instanceof MutableLogEvent) {
-                ((MutableLogEvent) other).initializeBuilder(this);
+            if (other instanceof ReusableLogEvent) {
+                ((ReusableLogEvent) other).initializeBuilder(this);
```

Comment:
It is nice to see those `instanceof` disappear, although it would be better to have a centralized logic on how to initialize the builder.